backend/src/main/java/nl/itvitae/twitbook/user/UserController.java

```diff
@@ -20,19 +24,46 @@
 public class UserController {
 
   private final UserRepository userRepository;
+  private final FollowRepository followRepository;
 
   @GetMapping
   public List<UserDTO> getAll() {
     return userRepository.findAll().stream().map(UserDTO::new).toList();
   }
 
   @GetMapping("by-username/{username}")
-  public ResponseEntity<?> findByUsername(@PathVariable String username) {
+  public ResponseEntity<?> findByUsername(@PathVariable String username,  @AuthenticationPrincipal User authUser) {
     Optional<User> user = userRepository.findByUsernameIgnoreCase(username);
 
-    if(user.isEmpty())
+    if (user.isEmpty()) {
       return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
 
-    return new ResponseEntity<>(new UserDTO(user.get()), HttpStatus.OK);
+    if (authUser == null) {
+      return new ResponseEntity<>(new UserDTO(user.get()), HttpStatus.OK);
+    }
+    return new ResponseEntity<>(new UserDTO(user.get(), followRepository.existsFollowByFollowerIdAndFollowingId(authUser.getId(), user.get().getId())), HttpStatus.OK);
+  }
+
+  @PostMapping("/follow/{followingUsername}")
+  public ResponseEntity<?> followUser(@PathVariable String followingUsername,
+      @AuthenticationPrincipal User user) {
+    Optional<User> following = userRepository.findByUsernameIgnoreCase(
+        followingUsername);
+    if (following.isEmpty()) {
+      return ResponseEntity.notFound().build();
+    }
+
+    if (user.getUsername().equals(followingUsername)) {
+      return new ResponseEntity<>(null, HttpStatus.CONFLICT);
+    }
+    Optional<Follow> optionalFollow = followRepository.findFollowByFollowerIdAndFollowingId(
```

Comment:
`existingFollow` would be a clearer description, maybe also a comment stating this code deletes any existing follows